microsoft/testsuites/core/debug.py

```diff
@@ -0,0 +1,118 @@
+        2. Check if /sys/kernel/debug/tracing/ is mounted, if not, mount it.
+        3. Get origin values of /sys/kernel/debug/tracing/kprobe_events and
+         /sys/kernel/debug/tracing/events/kprobes/my/enable.
+        4. Write "p:my filp_close" to /sys/kernel/debug/tracing/kprobe_events and
+         write "1" to /sys/kernel/debug/tracing/events/kprobes/my/enable.
+        5. Check if /sys/kernel/debug/tracing/kprobe_events and
+         /sys/kernel/debug/tracing/events/kprobes/my/enable are changed.
+        6. Write origin values back to /sys/kernel/debug/tracing/kprobe_events and
+         /sys/kernel/debug/tracing/events/kprobes/my/enable.
+
+        """,
+        priority=1,
+        requirement=simple_requirement(supported_platform_type=[AZURE, READY]),
+    )
+    def verify_enable_kprobe(self, node: Node) -> None:
+        if node.tools[KernelConfig].is_enabled("CONFIG_KPROBE_EVENTS"):
+            mount = node.tools[Mount]
+            if not (
+                mount.check_mount_point_exist("/sys/kernel/tracing")
+                or mount.check_mount_point_exist("/sys/kernel/debug")
+            ):
+                mount.mount("nodev", "/sys/kernel/debug", FileSystem.tracefs)
+
+            origin_kprobe_events: str = ""
+            origin_kprobe_enable: str = "0"
+            try:
+                cat = node.tools[Cat]
+
+                origin_kprobe_events = cat.read(
+                    "/sys/kernel/debug/tracing/kprobe_events", force_run=True, sudo=True
+                )
+                if origin_kprobe_events:
+                    origin_kprobe_enable = cat.read(
+                        "/sys/kernel/debug/tracing/events/kprobes/my/enable",
+                        force_run=True,
+                        sudo=True,
+                    )
+
+                echo = node.tools[Echo]
+                echo.write_to_file(
+                    "p:my filp_close",
+                    node.get_pure_path("/sys/kernel/debug/tracing/kprobe_events"),
+                    sudo=True,
+                )
+                echo.write_to_file(
+                    "1",
+                    node.get_pure_path(
+                        "/sys/kernel/debug/tracing/events/kprobes/my/enable"
+                    ),
+                    sudo=True,
+                )
+
+                cat = node.tools[Cat]
+                assert_that(
+                    cat.read(
+                        "/sys/kernel/debug/tracing/kprobe_events",
+                        sudo=True,
+                        force_run=True,
+                    )
+                ).is_equal_to("p:kprobes/my filp_close")
+                assert_that(
+                    cat.read(
+                        "/sys/kernel/debug/tracing/events/kprobes/my/enable",
+                        sudo=True,
+                        force_run=True,
+                    )
+                ).is_equal_to("1")
+            except Exception as ex:
```

Comment:
It do nothing, so can be totally removed. try...finally is enough.

Reply:
updated